0000-overhaul-qiskit-experiments.md

```diff
@@ -0,0 +1,638 @@
+although running analysis only requires experiment results (such as count dictionary) and metadata involving the setting of experiment.
+The interface to the experiment service is not necessary at all for analysis, and thus we can define more lightweight container.
+
+
+### Improvements in unit analysis
+
+Since 1-D parameter scan followed by a curve fitting is the most popular pattern in our experiments, QE provides the [`CurveAnalysis`](https://github.com/Qiskit-Extensions/qiskit-experiments/blob/main/qiskit_experiments/curve_analysis/curve_analysis.py) baseclass,
+so that an experiment developer can quickly write new analysis only by defining a fitting model and initial guess for the model parameters.
+Indeed, most calibration and maintenance experiments subclass the curve analysis baseclass. 
+That is to say, even small improvements will appear as a cumulative performance improvements in the heavily batched experiments.
+
+The subroutines with the significant execution time in the curve analysis are the curve fitter and figure drawer.
+QE uses the [LMFIT](https://lmfit.github.io/lmfit-py/) for curve fitting, which is basically a wrapper of the [Scipy Optimize](https://docs.scipy.org/doc/scipy/tutorial/optimize.html).
+By default, we use the `least_square` minimizer to minimize the residual of the fit curve.
+This optimizer is not quite slow, but it still requires numerically computed Jacobian matrix for parameter update.
+[JAX](https://jax.readthedocs.io/en/latest/index.html) provides a framework for JIT compilation and auto-gradient of a numpy compatible cost function.
+This allows us to choose more sophisticated optimizer such as BFGS, along with much faster residual computation.
+We found that this approach brought [more than x10 performance gain](https://github.com/Qiskit-Extensions/qiskit-experiments/issues/1192#issuecomment-1582050036) in the curve fitting.
+
+In contrast to the fitting, there is no straightforward way to reduce the drawer overhead other than lazy figure creation.
+At a scale of 1000 qubit experiment, maybe human experimentalist doesn't have a chance to visually confirm every experiment result.
+We could conditionally create a figure for unexpected experiment result (e.g. very poor chi-squared value, too short T1 value for T1 analysis) 
+to help a human experimentalist to scrutinize the device issue. Otherwise, we could create the result figure on the fly when it is accessed.
+
+### Proposal of new workflow
+
+Following sequence diagram proposes new workflow of QE run. 
+Note that now we have four local components, `Experiment`, `Analysis`, `Executor`, and `ExperimentData`.
+The experiment data becomes a data container which also implements the interface to the experiment service (this feature can be mix-in).
+Executor has two sub-executors `JobExecutor` and `AnalysisExecutor`.
```

Comment:
Potential confusion with the backend's JobExecutor, thus I'd rename it to `ExperimentExecutor`, or `ExperimentRunner` if it comes to be a 1st-class citizen type in QE.

Reply:
That's good point. I replaced it with `CircuitExecutor`, which aligns with circuit-runner runtime working behind the scene. Note that `ExperimentRunner` is `Executor` itself, because experiment = job + analysis execution.

[3382f99](https://github.com/Qiskit/RFCs/pull/47/commits/3382f9995293fc30063b3fbe2b1edb3b4edad797)